SubmissionTest.java

```diff
@@ -148,15 +148,21 @@ public void testZip() throws IOException
         
         // Get actual Submission filenames.
         File actualDir = new File("zips");
-        File[] actualZips = actualDir.listFiles();
-        Arrays.sort(actualZips);
-        String[] actual = new String[actualZips.length];
-        for (int i = 0; i < actual.length; i++)
+        
+        // Make sure directory has not been deleted 
+        // by deleteZips test or application. 
+        if (actualDir.exists())
         {
-            actual[i] = actualZips[i].getName();
-        }
+            File[] actualZips = actualDir.listFiles();
+            Arrays.sort(actualZips);
+            String[] actual = new String[actualZips.length];
+            for (int i = 0; i < actual.length; i++)
+            {
+                actual[i] = actualZips[i].getName();
+            }
 
-        assertArrayEquals("should be same", expected, actual);
+            assertArrayEquals("should be same", expected, actual);
```

Comment:
I don't think that we should have this assertion inside of the if statement.  Doesn't that mean that the test may or may not reach the assertion, making it potentially useless?